portal-impl/src/com/liferay/portlet/internal/PortletURLImpl.java

```diff
@@ -1520,7 +1527,7 @@ else if (_lifecycle.equals(PortletRequest.RENDER_PHASE)) {
 						}
 					}
 				}
-				else {
+				else if (_useNamespace) {
```

Comment:
I don't believe we should be touching the private render param namespace.  Is this change required for your solution to work?